t/plugin/ocsp-stapling.t

```diff
@@ -0,0 +1,676 @@
+--- response_body eval
+qr/OCSP response: no response sent/
+--- error_log
+no 'ocsp_stapling' field found, no need to run ocsp-stapling plugin
+
+
+
+=== TEST 6: client hello without status request extension required when enabled ocsp-stapling plugin
+--- config
+location /t {
+    content_by_lua_block {
+        local core = require("apisix.core")
+        local t = require("lib.test_admin")
+
+        local ssl_cert = t.read_file("t/certs/ocsp/rsa_good.crt")
+        local ssl_key =  t.read_file("t/certs/ocsp/rsa_good.key")
+
+        local data = {
+            cert = ssl_cert,
+            key = ssl_key,
+            sni = "ocsp.test.com",
+            ocsp_stapling = {
+                enabled = true
+            }
+        }
+
+        local code, body = t.test('/apisix/admin/ssls/1',
+            ngx.HTTP_PUT,
+            core.json.encode(data)
+        )
+
+        if code >= 300 then
+            ngx.status = code
+            ngx.say(body)
+            return
+        end
+
+        ngx.say(body)
+    }
+}
+--- response_body
+passed
+
+
+
+=== TEST 7: hit, handshake ok and no ocsp response send
+--- exec
+echo -n "Q" | $OPENSSL_BIN s_client -connect localhost:1994 -servername ocsp.test.com 2>&1 | cat
+--- response_body eval
+qr/CONNECTED/
+--- error_log
+no status request required, no need to send ocsp response
+
+
+
+=== TEST 8: cert without ocsp supported when enabled ocsp-stapling plugin
+--- config
+location /t {
+    content_by_lua_block {
+        local core = require("apisix.core")
+        local t = require("lib.test_admin")
+
+        local ssl_cert = t.read_file("t/certs/apisix.crt")
+        local ssl_key =  t.read_file("t/certs/apisix.key")
+
+        local data = {
+            cert = ssl_cert,
+            key = ssl_key,
+            sni = "test.com",
+            ocsp_stapling = {
+                enabled = true
+            }
+        }
+
+        local code, body = t.test('/apisix/admin/ssls/1',
+            ngx.HTTP_PUT,
+            core.json.encode(data)
+        )
+
+        if code >= 300 then
+            ngx.status = code
+            ngx.say(body)
+            return
+        end
+
+        ngx.say(body)
+    }
+}
+--- response_body
+passed
+
+
+
+=== TEST 9: hit, handshake ok:1
+--- exec
+echo -n "Q" | $OPENSSL_BIN s_client -connect localhost:1994 -servername test.com -status 2>&1 | cat
+--- response_body eval
+qr/CONNECTED/
+--- error_log
+no ocsp response send: failed to get ocsp url: cert not contains authority_information_access extension
+
+
+
+=== TEST 10: hit, no ocsp response send due to get ocsp responder url failed:2
+--- exec
+echo -n "Q" | $OPENSSL_BIN s_client -connect localhost:1994 -servername test.com -status 2>&1 | cat
+--- response_body eval
+qr/OCSP response: no response sent/
+--- error_log
+no ocsp response send: failed to get ocsp url: cert not contains authority_information_access extension
+
+
+
+=== TEST 11: run ocsp responseder, will exit when test finished
```

Comment:
What does `responseder` mean?

Reply:
It use for receive ocsp requests and response it.

Reply:
is it a typo?

Reply:
yes, I will fix it.